Nitrox.Model/Helper/NetHelper.cs

```diff
@@ -56,17 +65,43 @@ public static IEnumerable<NetworkInterface> GetInternetInterfaces() =>
             }
         }
 
+        IPAddress? ipv6Candidate = null;
         foreach (NetworkInterface ni in GetInternetInterfaces())
         {
             foreach (UnicastIPAddressInformation ip in ni.GetIPProperties().UnicastAddresses)
             {
-                if (ip.Address.AddressFamily == AddressFamily.InterNetwork)
+                IPAddress address = ip.Address;
+                if (address.IsIPv4MappedToIPv6)
+                {
+                    IPAddress normalized = NormalizeAddress(address);
+                    lock (lanIpLock)
+                    {
+                        return lanIpCache = normalized;
+                    }
+                }
+                if (address.AddressFamily == AddressFamily.InterNetwork)
                 {
                     lock (lanIpLock)
                     {
-                        return lanIpCache = ip.Address;
+                        return lanIpCache = NormalizeAddress(address);
                     }
                 }
+
+                if (ipv6Candidate == null
+                    && address.AddressFamily == AddressFamily.InterNetworkV6
+                    && !address.IsIPv6LinkLocal
+                    && !address.IsIPv6Multicast)
+                {
+                    ipv6Candidate = address;
```

Comment:
Unless I have missed something why would you want to continue looping after assigning here rather than just returning early after assigning the cache